Homework9/Hw9/Services/ExpressionParser/ExpressionParserService.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System.Linq.Expressions;
+using static Hw9.ErrorMessages.MathErrorMessager;
+
+namespace Hw9.Services.ExpressionParser
+{
+    public class ExpressionParserService : IExpressionParserService
+    {
+        public const string Plus = "+";
+        public const string Minus = "-";
+        public const string Multiply = "*";
+        public const string Divide = "/";
+        public const string OpenBracket = "(";
+        public const string CloseBracket = ")";
+        public const string UnaryMinus = "_";
+
+        public bool CheckExpressionString(string? expression, out string errorMessage)
```

Comment:
```suggestion
      private bool CheckExpressionString(string expression, out string errorMessage)
```
использовать как внутренний метод парсера.
Текст ошибки можно вернуть ошибкой (например, создать MathExpressionParseException(message) и ловить его), создать монаду Result<Expression> с сообщением от ошибки. Если свою монаду не делать, то можно использовать библиотеку OneOf, по сути, являющейся монадой